lib/core/test/unit/Cardano/Wallet/NetworkSpec.hs

```diff
@@ -37,5 +58,44 @@ spec = do
         testShow $ ErrPostTxBadRequest mempty
         testShow $ ErrPostTxProtocolFailure mempty
 
+    followSpec
+
+followSpec :: Spec
+followSpec =
+    describe "follow" $ do
+        it "retries on exceptions, but not too often" $ do
+            tvar <- newTVarIO []
+            let tr = traceInTVarIO tvar
+            let getHeader = header
+            let advance _blocks _h = return $ Continue @()
+            let rollback _slot = return $ Continue @()
+            void $ race (threadDelay $ 10 * second)
+                (follow mockNetworkLayer tr [] advance rollback getHeader)
+            errors <- mapMaybe (unMsg . loContent) <$> readTVarIO tvar
+            case length errors of
+                x | x == 4 -> return ()
+                  | otherwise -> expectationFailure
+                        $ "we expected 4 errors to be logged, not " ++ show x
```

Comment:
I used `expectationFailure` instead of `shouldBe` specifically to keep the test description more generic while still being clear that we expect 4 errors.